Sources/Plasma/Apps/plClient/CMakeLists.txt

```diff
@@ -90,17 +89,102 @@ if(WIN32)
         win32/res/Dirt.ICO
         win32/res/headspin.ico
     )
+elseif(APPLE)
+    set(plClient_SOURCES ${plClient_SOURCES}
+        Mac-Cocoa/main.mm
+        Mac-Cocoa/PLSKeyboardEventMonitor.mm
+        Mac-Cocoa/PLSView.mm
+        Mac-Cocoa/PLSLoginWindowController.mm
+        Mac-Cocoa/PLSPatcherWindowController.mm
+        Mac-Cocoa/PLSPatcher.mm
+        Mac-Cocoa/PLSServerStatus.mm
+        Mac-Cocoa/StringTheory_NSString.mm
+    )
+    set(plClient_HEADERS ${plClient_HEADERS}
+        Mac-Cocoa/PLSKeyboardEventMonitor.h
+        Mac-Cocoa/PLSView.h
+        Mac-Cocoa/PLSLoginWindowController.h
+        Mac-Cocoa/PLSPatcherWindowController.h
+        Mac-Cocoa/PLSPatcher.h
+        Mac-Cocoa/PLSServerStatus.h
+        Mac-Cocoa/StringTheory_NSString.h
+    )
+    set(RESOURCES 
+        Mac-Cocoa/MainMenu.xib
+        Mac-Cocoa/PLSLoginWindowController.xib
+        Mac-Cocoa/PLSPatcherWindowController.xib
+        Mac-Cocoa/banner@2x.jpeg
+        win32/res/Dirt.ICO
+    )
+    #shaders need to be compiled as part of the app
+    #this could change in the future, but for now the Metal code expects the library to be compiled in the app
+    set(plClient_SHADERS
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/FixedPipelineShaders.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/PlateShaders.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/BiasNormals.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/CompCosines.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveSet7.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/Grass.metal
+	../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveDecEnv.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/Avatar.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveDec1Lay_7.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/WaveRip.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/Clear.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/GammaCorrection.metal
+    ../../FeatureLib/pfMetalPipeline/ShaderSrc/TextFontShader.metal
+    )
+    set_source_files_properties(${plClient_SHADERS} PROPERTIES LANGUAGE METAL)
+    source_group("Metal Shaders" FILES ${plClient_SHADERS})
```

Comment:
I would be a little bit nervous about doing this before the files are added to the CMake target.